ydb/core/blobstorage/dsproxy/dsproxy_get_block.cpp

```
@@ -0,0 +1,139 @@
+#include <ydb/core/blobstorage/base/blobstorage_events.h>
+#include <ydb/core/blobstorage/vdisk/common/vdisk_events.h>
+
+namespace NKikimr {
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// GET BLOCK request
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+class TBlobStorageGroupGetBlockRequest : public TBlobStorageGroupRequestActor {
+    const ui64 TabletId;
+    ui64 Generation;
+    const TInstant Deadline;
+    ui64 Requests;
+    ui64 Responses;
+    TGroupQuorumTracker QuorumTracker;
+
+    void Handle(TEvBlobStorage::TEvVGetBlockResult::TPtr &ev) {
+        ProcessReplyFromQueue(ev->Get());
+        const NKikimrBlobStorage::TEvVGetBlockResult &record = ev->Get()->Record;
+        Y_ABORT_UNLESS(record.HasStatus());
+        const NKikimrProto::EReplyStatus status = record.GetStatus();
+        Y_ABORT_UNLESS(record.HasVDiskID());
+        const TVDiskID vdisk = VDiskIDFromVDiskID(record.GetVDiskID());
+
+        DSP_LOG_LOG_S(PriorityForStatusInbound(status), "DSPGB01", "Handle TEvVGetBlockResult"
+            << " status# " << NKikimrProto::EReplyStatus_Name(status).data()
+            << " From# " << vdisk.ToString()
+            << " NodeId# " << Info->GetActorId(vdisk).NodeId());
+
+        if (record.HasGeneration()) {
+            const ui64 generation = record.GetGeneration();
+            Generation = Max(Generation, generation);
+        }
+        ++Responses;
+
+        switch (const NKikimrProto::EReplyStatus overallStatus = QuorumTracker.ProcessReply(vdisk, status)) {
+            case NKikimrProto::OK:
+                if (Responses == Requests) {
+                    ReplyAndDie(NKikimrProto::OK);
+                }
+                break;
+
+            case NKikimrProto::ERROR:
+                ReplyAndDie(NKikimrProto::ERROR);
+                break;
+
+            default:
```

not sure other cases should be silently dropped